tests/integration/Core/Content/Product/ProductEntityTest.php

```diff
@@ -0,0 +1,73 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Tests\Integration\Core\Content\Product;
+
+use PHPUnit\Framework\Attributes\DataProvider;
+use PHPUnit\Framework\TestCase;
+use Shopware\Core\Content\Product\ProductEntity;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Test\TestCaseBase\IntegrationTestBehaviour;
+use Twig\Loader\ArrayLoader;
+
+/**
+ * @internal
+ */
+#[Package('inventory')]
+class ProductEntityTest extends TestCase
+{
+    use IntegrationTestBehaviour;
+
+    #[DataProvider('productProvider')]
+    public function testStringifyProduct(ProductEntity $entity, string $expected): void
+    {
+        static::assertEquals($expected, $this->renderTestTemplate($entity));
```

Comment:
```suggestion
        static::assertSame($expected, $this->renderTestTemplate($entity));
```
please use `assertSame` where possible instead, as it is type safe